packages/twenty-server/src/engine/api/graphql/workspace-query-builder/factories/foreign-data-wrapper-query.factory.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+
+import {
+  FdwOptions,
+  RemoteServerType,
+  UserMappingOptions,
+} from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+
+@Injectable()
+export class ForeignDataWrapperQueryFactory {
+  createFDW(
+    fdwId: string,
+    fdwType: RemoteServerType,
+    fdwOptions: FdwOptions<RemoteServerType>,
+  ) {
+    let fdwName = '';
+    let options = '';
+
+    switch (fdwType) {
+      case RemoteServerType.POSTGRES_FDW:
+        fdwName = 'postgres_fdw';
+        options = this.buildPostgresFDWQueryOptions(fdwOptions);
+        break;
+      default:
+        throw new Error('FDW type not supported');
```

Comment:
I would prefer that we throw business errors: new ForeignDataWrapperTypeUnsupported and we catch it in the controller and throw it as 400. For this one, we should also add a check on GraphQL type itself direclty